app/services/catalog/add_to_order.rb

```diff
@@ -9,7 +9,11 @@ def initialize(params)
     def process
       order = Order.find_by!(:id => @params[:order_id])
       @order_item = order.order_items.create!(order_item_params)
+
       self
+    rescue ActiveRecord::RecordInvalid => e
```

Comment:
@eclarizio wouldn't this be already handled by the rescue from in the common gem.
https://github.com/RedHatInsights/insights-api-common-rails/blob/e4df2e11ff19496038fe9c66067e275e4554e5ce/lib/insights/api/common/application_controller_mixins/exception_handling.rb#L10
Is the common gem just missing a log line, since it always ends up creating a json payload to send back to the callers.

Reply:
Good point, that's fine by me if we can get it in there, then.

Reply:
@eclarizio A nice enhancement would be to have an error log which logs only the errors similar to how Apache does it. If the error.log is empty then we can safely assume that no errors have been generated and the system is working without any issues. If we see the error.log growing then we know that the system is having issues.

Reply:
@mkanoor for the cloud services this is not really an option.  We are not collecting the logs in a local file.  Maybe cloud_watch gives us another way to do what you want but more likely we would want to instrument some logic to monitor the services and alert on certain conditions.  Basically, I think there is a different way to approach this.